shared/teams/team/members/container.js

```diff
@@ -0,0 +1,66 @@
+// @flow
+import * as Constants from '../../../constants/teams'
+import * as Types from '../../../constants/types/teams'
+import * as I from 'immutable'
+import {type TypedState, connect} from '../../../util/container'
+import {Members} from '.'
+
+export type OwnProps = {
+  teamname: string,
+}
+
+const order = {owner: 0, admin: 1, writer: 2, reader: 3}
+
+const getOrderedMemberArray = (
+  memberInfo: I.Set<Types.MemberInfo>,
+  you: ?string,
+  listYouFirst: boolean
+): Array<Types.MemberInfo> => {
+  let youInfo
+  let info = memberInfo
+  if (you && !listYouFirst) {
+    youInfo = memberInfo.find(member => member.username === you)
+    if (youInfo) {
+      info = memberInfo.delete(youInfo)
+    }
+  }
+  let returnArray = info
+    .toArray()
+    .sort(
+      (a, b) =>
+        !a.type || !b.type || a.type === b.type
```

Comment:
this bug may have already existed, but this isn't a consistent sort function, since it isn't antisymmetric (a <= b and b <= a implies a equiv to b), and it isn't transitive (a <= b and b <= c implies a <= c).

For a counterexample to antisymmetry, let `a = { name: 'foo' }` and `b = { type: 'reader', name: 'foo' }`

For a counterexample to transitivity, let `a = { name: 'foo' }`, b = `{ type: 'owner', name: 'goo' }`, `c = { type: 'reader', name: 'bar' }`

It suffices to have `unknown: 4` in `order`, then set the type for a missing `type` to be `unknown`, and sort by type first then by name.

Reply:
I didn't want to mutate the data coming in, so I replaced the sort with a more explicit check hierarchy that I think preserves transitivity & antisymmetry. PTAL